src/otx/core/model/entity/classification.py

```diff
@@ -75,12 +80,61 @@ def head_forward_fn(self, x: torch.Tensor) -> torch.Tensor:
 
     def remove_explain_hook_handle(self) -> None:
         """Removes explain hook from the model."""
-        if self.explain_hook.handle is not None:
+        if self.explain_hook is not None and self.explain_hook.handle is not None:
             self.explain_hook.handle.remove()
 
+    def remove_explain_hook(self) -> None:
+        """Removes explain hook completely."""
+        self.remove_explain_hook_handle()
+        self.explain_hook = None
+
     def reset_explain_hook(self) -> None:
         """Clear all history of explain records."""
-        self.explain_hook.reset()
+        if self.explain_hook is not None:
+            self.explain_hook.reset()
+
+    @property
+    def _exportable_model(self) -> nn.Module:
+        """nn.Module which will be passed to the exporter."""
+        if self.explain_hook is not None:
+
+            class ModelWithExplainOutput(nn.Module):
+                def __init__(self, model: nn.Module, hook: BaseRecordingForwardHook) -> None:
+                    super().__init__()
+                    self.model = model
+                    self.hook = hook
+                    self.hook.reset()
+
+                def forward(self, x: torch.Tensor) -> tuple:
```

Comment:
The final mechanism of obtaining the saliency maps is still not clear. Actually, we can already access the hook in `_customize_outputs`, the problem is what to do with the raw saliency maps. This wrapper doesn't look as a comprehensive replacement to forward, it's rather a workaround if batch is 1.

Reply:
Also, if we look in general, the whole idea of hooks is that we don't change anything in the forward itself, while can extend it with an arbitrary number of hooks. 
If we want a custom forward, we don't need hooks, but with hooks we have to generate a custom forward, since torch tracer is quite limited in terms of functionality.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/2870#discussion_r1475416405
https://github.com/openvinotoolkit/training_extensions/pull/2900#discussion_r1487040737
Like these comments, I think that we should introduce a dedicated `forward()` function for XAI at this time.

Reply:
@vinnamkim does that mean we have to remove the hook? If we have a standalone forward, keeping the hooks makes no sense, like I described in the previous comment.

Reply:
I guess you guys can move forward gradually. For example,

1. Introduce `forward_explain()` with the hook

```python
def forward_explain(self, inputs) -> BatchPredictionWithXAI:
     handle = self.register_explain_hook()
     outputs = self.forward(inputs)
     augmented_outputs = augment_with_xai_outputs(outputs, handle.xai_outputs)
     handle.remove()
     return augmented_outputs
```
2. Remove using the hook from `forward_explain()`